e2e-tests/tfm/cypress/e2e/journeys/deal-cancellation/check-details.spec.js

```diff
@@ -32,29 +32,39 @@ context('Deal cancellation - check details', () => {
     });
   });
 
-  describe('when logged in as a PIM user', () => {
-    beforeEach(() => {
-      cy.login(PIM_USER_1);
-      cy.visit(relative(`/case/${dealId}/deal`));
+  beforeEach(() => {
+    cy.login(PIM_USER_1);
+    cy.visit(relative(`/case/${dealId}/deal`));
 
-      caseDealPage.cancelDealButton().click();
+    caseDealPage.cancelDealButton().click();
 
-      cy.url().should('eq', relative(`/case/${dealId}/cancellation/reason`));
-      cy.clickContinueButton();
+    cy.url().should('eq', relative(`/case/${dealId}/cancellation/reason`));
+    cy.clickContinueButton();
 
-      cy.url().should('eq', relative(`/case/${dealId}/cancellation/bank-request-date`));
+    cy.url().should('eq', relative(`/case/${dealId}/cancellation/bank-request-date`));
 
-      cy.completeDateFormFields({ idPrefix: 'bank-request-date' });
+    cy.completeDateFormFields({ idPrefix: 'bank-request-date' });
 
-      cy.clickContinueButton();
+    cy.clickContinueButton();
 
-      cy.url().should('eq', relative(`/case/${dealId}/cancellation/effective-from-date`));
+    cy.url().should('eq', relative(`/case/${dealId}/cancellation/effective-from-date`));
+
+    cy.completeDateFormFields({ idPrefix: 'effective-from-date', date: yesterday.date });
+
+    cy.clickContinueButton();
+  });
 
-      cy.completeDateFormFields({ idPrefix: 'effective-from-date', date: tomorrow.date });
+  describe('when logged in as a non-PIM user', () => {
+    beforeEach(() => {
+      cy.login(T1_USER_1);
+    });
 
-      cy.clickContinueButton();
+    it('should redirect when visiting the check details page', () => {
+      cy.url().should('eq', relative('/deals/0'));
     });
+  });
 
+  describe('when logged in as a PIM user', () => {
```

Comment:
It was doing from the top level `beforeEach` but I've refactored to make it clearer whats happening now. Previously the `'when logged in as a non-PIM user'` test wasnt actually visiting the correct page (and wouldnt be able to unless a PIM user has previously created a cancellation)

Reply:
The dificulty in testing the `non-pim user` does make me think this should just be covered by api-tests (though we dont have any for tfm-ui)

Reply:
Thanks @MarRobSoftwire I think the non-pim user should be in it's own E2E test so we can test it fully. I do not think it should only be covered by unit tests since there is different functionality for different user types